src/tempdir.js

```diff
@@ -0,0 +1,39 @@
+const output = require('./output')
+const ostmpdir = require('os-tmpdir')
+const fs = require('fs')
+
+/**
+ * Method to determine where to save devlab.sh tmp file
+ * @returns {String} path of writeable temp dir
+ */
+module.exports = () => {
+  /**
+   * Runs sync access for W_OK to determine if we can write to the path
+   * @params {String} path The path to test
+   * @returns {Boolean}
+   */
+  const isWriteable = (path) => {
+    try {
+      fs.accessSync(path, fs.W_OK)
+      return true
+    } catch (e) {
+      return false
+    }
+  }
+
+  // Check our options...
+  if (process.env.DEVLAB_TMP && isWriteable(process.env.DEVLAB_TMP)) {
+    // Use user-specified DEVLAB_TMP
+    return process.env.DEVLAB_TMP
+  } else if (isWriteable('/tmp')) {
+    // Use (most common) /tmp
+    return '/tmp'
+  } else if (isWriteable(require('os-tmpdir')())) {
```

Comment:
Ah ok. I do remember having odd trouble with `os-tmpdir` in the past, which seemed silly since the whole point of that module is that it should _just work_. Thanks for the explanation.